common/flusher/flusher_test.go

```diff
@@ -0,0 +1,117 @@
+// The MIT License
+//
+// Copyright (c) 2022 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package flusher
+
+import (
+	"context"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"go.temporal.io/server/common/future"
+	"go.temporal.io/server/common/log"
+	"sync"
+	"testing"
+)
+
+type (
+	flusherSuite struct {
+		*require.Assertions
+		suite.Suite
+		controller *gomock.Controller
+		ctx        context.Context
+
+		capacity int
+		sync.Mutex
+		flusher *Flusher[int]
+	}
+)
+
+func TestFlusher(t *testing.T) {
```

Comment:
add some concurrency test if possible

Reply:
Are there any example concurrency tests in the repo that you can point me to? I've run tests with -race condition if that helps

Reply:
E.g. https://github.com/temporalio/temporal/blob/master/common/backoff/retry_test.go#L147